wandb/media.py

```diff
@@ -0,0 +1,81 @@
+import os
+from wandb import util
+
+
+class Media(object):
+    @staticmethod
+    def from_type(type):
+        pass
+
+
+class Image(object):
+    def __init__(self, data, mode=None, caption=None):
+        """
+        Accepts numpy array of image data, or a PIL image. The class attempts to infer
+        the data format and converts it.
+        """
+        try:
+            from PIL import Image as PILImage
+        except ImportError:
+            raise ValueError(
```

Comment:
Cool I like this behavior of only requiring certain dependencies at runtime, although that raises the question. Do we put it in requirements.txt?

I think when you pip install a package you can pass arguments. So we could maybe have like pip install wandb --full or pip install wandb --barebones. Not sure though.

Reply:
Yeah, I've seen pip install urllib3[secure] maybe we could do something like that but for PyTorch, keras, TensorFlow, etc.